### ISSUE TITLE
New wikitext syntax for representing literal text

### ISSUE THREAD

**Issue by Jermolene** [Jermolene/TiddlyWiki5#1473]
Sometimes it would be useful to be able to use character sequences like `//` without triggering the corresponding wikitext syntax. It is proposed to introduce a new wikitext parse rule that marks plain, literal text that should not be further parsed.

For example:

```
===this is //not// italicised===
```

**Comment by grayeul** [comment]
Yep, that would be great.... but the obvious question follows, how do you put '===' into plain text mode?

**Comment by Jermolene** [comment]
It's weird and clumsy but you could do it like this:

```
===plain===&#61;&#61;&#61;===plain again===
```

**Comment by grayeul** [comment]
OK -- clumsy is better than not doable.  I guess I didn't realize or think about solving the // problem with just putting in the ascii codes for that, which would have been the quick solution.  However, I do think having some way to mark regions not to be  parsed is helpful.  But is there objection to also using something (like a backslash) to escape only the next character?  This is very typical in the unix shell world....  If that worked, you could probably do:
=== All of this is plain including the \=== three equal signs to the left ===
You could potentially escape each of the three equal signs, but it may be that if the first one is escaped, then two subsequent equal signs don't mean anything (unless that is other wikitext I don't know about)

**Comment by tobibeer** [comment]
:+1:  this will likely also fix #1445

**Comment by pmario** [comment]
```
===this is //not// italicised===
```

That's wrong, because `not` is not italicised :)) (just ignore this sentence. I couldn't withstand :)

IMO `~` is the marker, that prevents wikitext rendering. so may be `~//not//` would be an option, if the parser can handle it.

**Comment by Jermolene** [comment]
> But is there objection to also using something (like a backslash) to escape only the next character?

> IMO `~` is the marker, that prevents wikitext rendering. so may be `~//not//` would be an option, if the parser can handle it.

The problem with this proposal is that it does indeed complicate the parsing, and  it means that users would need to locate and treat each individual problem in a passage of text.

**Comment by grayeul** [comment]
Yes, I understand it would not be desirable to have this as the only way to address literal text, as it would be overly cumbersome.  I'm asking if it is reasonable/doable (obviously more work :)  to have an *additional* mechanism that just prevents the next character from being recognized as markup, and instead is taken as is.  I had suggested the backslash as this is common in the unix world, but as pmario suggests the tilde is really the character used (most closely) to that... but it seems to me that it would apply more to the following token, not just a character -- which isn't exactly the same thing.

**Comment by tobibeer** [comment]
I believe the main question is this: do we want literal text to be enclosed by certain markup (which I prefer) or somehow "escaped"... which then raises the problem of reverting the escaping.

**Comment by Jermolene** [comment]
> but it seems to me that it would apply more to the following token, not just a character -- which isn't exactly the same thing.

You're right: your proposal of acting on the next character is significantly easier to parse. The only issue is that this new parse rule would completely consume that character within wikitext: one wouldn't be able to use that character in most circumstances without being aware of the need to escape it. So, it would make it significantly harder to type single backslashes, in your example.

> I believe the main question is this: Do we want literal text to be enclosed by certain markup — which I prefer — or somehow "escaped"... which then raises the problem of reverting the escaping and also of having to do that with more than a single "word".

Yes, that's right. This proposal is specifically about the enclosed variant. We already use the escape prefix for linking, and we can consider separately whether we want to extend that.

**Comment by grayeul** [comment]
Ultimately someone (you) have to decide what the allowable rules are and what is what users have to be expected to know in order to type stuff in.  It is clear that the escape character (backslash for the sake of argument right now) would have to be known  -- that it *is* an escape character.  But the same can be said for any wiki markup.  i.e. when I was trying to enter text with two forward slashes:  ABC//OtherInfo -- it wasn't working for me because the // was special markup.  Granted, you are less likely to run into that scenario with a combination of characters (two or more) then with a single character -- but either way someone has to know.  In the unix world, if you need a real backslash, you just escape it -- so to get a single backslash '\'  -- you usually have to type in '\\' and if you wanted two, then it gets even ugliers:  '\\\\' --- but this is reasonably well-known.  

So, the question is can we find a good character to use for this, if there is adequate support for it. If the backslash is mainly used  with the \define and \end keywords, then (unless you could special case those, which you probably couldn't) using the backslash as an escape, would effectively change those to be:  \\define and \\end.... which is obnoxious, considering there are plenty of macro definitions in place, but possibly localized to small, easy to find area.  Maybe there are other characters more suited... but surely there will almost always be *someone* who will be surprised that they can't just type arbitrary character 'X' without having to do something special.

Anyway -- it sounds like overall having a character "completely consumed" within wikitext, is probably too big a pill to swallow.  And I'm ok without having the single-character escape option available, falling back to the previous suggest as '====' as a pre/post marker for literal text is probably helpful enough.  But I appreciate the discussion.

**Comment by tobibeer** [comment]
I think the best way to move forward is to collapse all the discussion above and move forward with the OP.

(A global escape character perhaps introduces more problems than it solves.)

**Comment by buggyj** [comment]
We could have a 'nocharacter character' - something that parser to nothing. At present 4 backticks give a space - we could  make it null so  "/````/no italics"  give "//noitalics" instead of "/ /noitalics"

**Comment by tobibeer** [comment]
@buggyj, the point of the exercise is to never touch the text that is to be rendered as is, so these kinds of escaping mechanisms are not an option

**Comment by buggyj** [comment]
Actually this is already possible with four single quots /''''/ renders as //

**Comment by buggyj** [comment]
maybe 
$$$ this is plain text $$$ 
could be used as it already has this kind of use

**Comment by tobibeer** [comment]
Yes, but then it looks like code, and not like normal text.

**Comment by tobibeer** [comment]
Another markup could be four enclosing double quotes, because...

```
""""
this is not only //literal// ''text''
without any markup being rendered
but hard linebreaks would also apply, obviously
""""
```

This could potentially even be done by extending the existing parser.
